src/tests/src/solver/expressions/test_expressions.cpp

```diff
@@ -260,6 +279,90 @@ BOOST_FIXTURE_TEST_CASE(comparison_node, Registry<Node>)
     BOOST_CHECK_EQUAL(printed, "(22.000000-8.000000)>=(8.000000-22.000000)");
 }
 
+BOOST_AUTO_TEST_CASE(linear_status_plus)
+{
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR + LinearStatus::LINEAR, LinearStatus::LINEAR);
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR + LinearStatus::CONSTANT, LinearStatus::LINEAR);
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR + LinearStatus::NON_LINEAR, LinearStatus::NON_LINEAR);
+
+    BOOST_CHECK_EQUAL(LinearStatus::CONSTANT + LinearStatus::CONSTANT, LinearStatus::CONSTANT);
+    BOOST_CHECK_EQUAL(LinearStatus::CONSTANT + LinearStatus::NON_LINEAR, LinearStatus::NON_LINEAR);
+
+    BOOST_CHECK_EQUAL(LinearStatus::NON_LINEAR + LinearStatus::NON_LINEAR,
+                      LinearStatus::NON_LINEAR);
+}
+
+BOOST_AUTO_TEST_CASE(linear_status_mult)
+{
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR * LinearStatus::LINEAR, LinearStatus::NON_LINEAR);
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR * LinearStatus::CONSTANT, LinearStatus::LINEAR);
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR * LinearStatus::NON_LINEAR, LinearStatus::NON_LINEAR);
+
+    BOOST_CHECK_EQUAL(LinearStatus::CONSTANT * LinearStatus::CONSTANT, LinearStatus::CONSTANT);
+    BOOST_CHECK_EQUAL(LinearStatus::CONSTANT * LinearStatus::NON_LINEAR, LinearStatus::NON_LINEAR);
+
+    BOOST_CHECK_EQUAL(LinearStatus::NON_LINEAR * LinearStatus::NON_LINEAR,
+                      LinearStatus::NON_LINEAR);
+}
+
+BOOST_AUTO_TEST_CASE(linear_status_divide)
+{
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR / LinearStatus::LINEAR, LinearStatus::NON_LINEAR);
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR / LinearStatus::CONSTANT, LinearStatus::LINEAR);
+    BOOST_CHECK_EQUAL(LinearStatus::LINEAR / LinearStatus::NON_LINEAR, LinearStatus::NON_LINEAR);
+
+    BOOST_CHECK_EQUAL(LinearStatus::CONSTANT / LinearStatus::CONSTANT, LinearStatus::CONSTANT);
+    BOOST_CHECK_EQUAL(LinearStatus::CONSTANT / LinearStatus::NON_LINEAR, LinearStatus::NON_LINEAR);
+
+    BOOST_CHECK_EQUAL(LinearStatus::NON_LINEAR / LinearStatus::NON_LINEAR,
+                      LinearStatus::NON_LINEAR);
+}
+
+static const std::vector<LinearStatus> LinearStatus_ALL = {LinearStatus::LINEAR,
+                                                           LinearStatus::NON_LINEAR,
+                                                           LinearStatus::CONSTANT};
+
+BOOST_AUTO_TEST_CASE(linear_status_minus)
```

Comment:
Instead of loops for over the same list of value, could we not do like we did in infeasible problems unit tests : 
```C++
BOOST_DATA_TEST_CASE(linear_status_minus,
                     bdata::make(LinearStatus_ALL),
                     x)
{
    BOOST_CHECK_EQUAL(x, -x);
}
```
or something like that...

Reply:
Done 2ebb72192